### ISSUE TITLE
Thread running at QOS_CLASS_USER_INITIATED when using findAll()

### ISSUE THREAD

**Issue by vdkdamian** [netreconlab/Parse-Swift#13]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/netreconlab/Parse-Swift/security/policy).
- [x] I have searched through [existing issues](https://github.com/netreconlab/Parse-Swift/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Swift SDK](https://github.com/netreconlab/Parse-Swift/releases). 

### Issue Description
I am getting a hang risk when using findAll(). It's still working, but I thought I should mention it.

### Steps to reproduce
Using findAll()

### Actual Outcome
Thread warning

### Expected Outcome
n/a

### Environment

Client
- Parse Swift SDK version: `4.16.0`
- Xcode version: `14.1`
- Operating system (iOS, macOS, watchOS, etc.): `iOS, macOS`
- Operating system version: `16.1, 13.0`

Server
- Parse Server version: `4.5.0
- Operating system: `Back4App`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `AWS`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `3.6`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `AWS`

### Logs
```
Thread Performance Checker: Thread running at QOS_CLASS_USER_INITIATED waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions
PID: 62080, TID: 2639256
Backtrace
=================================================================
3   Ipegro                              0x000000010166a87c $s10ParseSwift3APIV03NonA11BodyCommandV7execute7optionsq_ShyAC6OptionOG_tKF + 1760
4   Ipegro                              0x000000010197e83c $s10ParseSwift5QueryV7findAll10batchLimit7options13callbackQueue10completionySiSg_ShyAA3APIV6OptionOGSo17OS_dispatch_queueCys6ResultOySayxGAA0A5ErrorVGctFyyXEfU0_ + 1080
5   Ipegro                              0x0000000101996040 $s10ParseSwift5QueryV7findAll10batchLimit7options13callbackQueue10completionySiSg_ShyAA3APIV6OptionOGSo17OS_dispatch_queueCys6ResultOySayxGAA0A5ErrorVGctFyyXEfU0_TA + 68
6   Ipegro                              0x000000010174dc0c $sIg_Ieg_TR + 20
7   Ipegro                              0x000000010174dc30 $sIeg_IyB_TR + 24
8   libdispatch.dylib                   0x000000010332e9d4 _dispatch_client_callout + 20
9   libdispatch.dylib                   0x0000000103343e98 _dispatch_sync_invoke_and_complete + 192
10  Ipegro                              0x000000010197e27c $s10ParseSwift5QueryV7findAll10batchLimit7options13callbackQueue10completionySiSg_ShyAA3APIV6OptionOGSo17OS_dispatch_queueCys6ResultOySayxGAA0A5ErrorVGctF + 2208
11  Ipegro                              0x000000010196644c $s10ParseSwift5QueryV7findAll10batchLimit7optionsSayxGSiSg_ShyAA3APIV6OptionOGtYaKFyScCyAGs5Error_pGXEfU_ + 204
12  libswift_Concurrency.dylib          0x0000000210c73d38 $ss31withCheckedThrowingContinuation8function_xSS_yScCyxs5Error_pGXEtYaKlFySccyxsAC_pGXEfU_TA + 164
13  libswift_Concurrency.dylib          0x0000000210c73e4c $ss30withUnsafeThrowingContinuationyxySccyxs5Error_pGXEYaKlF + 228
14  libswift_Concurrency.dylib          0x0000000210cae4c4 _ZN5swift34runJobInEstablishedExecutorContextEPNS_3JobE + 376
15  libswift_Concurrency.dylib          0x0000000210caf408 _ZL17swift_job_runImplPN5swift3JobENS_11ExecutorRefE + 72
16  libdispatch.dylib                   0x000000010334693c _dispatch_root_queue_drain + 408
17  libdispatch.dylib                   0x0000000103347560 _dispatch_worker_thread2 + 196
18  libsystem_pthread.dylib             0x0000000102f3bcec _pthread_wqthread + 228
19  libsystem_pthread.dylib             0x0000000102f43a7c start_wqthread + 8
```

[Screenshot](https://user-images.githubusercontent.com/44820440/200348939-91ac7475-8c3f-4d4b-940e-7efb66f242fd.png)

**Comment by cbaker6** [comment]
Good find, the PR should fix the warning
